test/e2e/framework/kcp.go

```diff
@@ -98,6 +99,16 @@ func newKcpServer(t *testing.T, cfg KcpConfig, artifactDir, dataDir string) (*kc
 	if err := os.MkdirAll(dataDir, 0755); err != nil {
 		return nil, fmt.Errorf("could not create data dir: %w", err)
 	}
+
+	saPrivateKey, err := GenerateRSAPrivateKey()
```

Comment:
Given that a default key is being set (and this test code isn't even supplying the result of generation as an argument to the server), this change and the rsa.go addition should likely be dropped.

Reply:
done